### PR TITLE
Feature/availability page

### PR DESCRIPTION
**Pull request by olena-kozak**
Create availability page

image
image

### CODE REVIEW

File: frontend/src/app/modules/availability/availability-page/availability-page.component.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+    selector: 'app-availability-page',
+    templateUrl: './availability-page.component.html',
+    styleUrls: ['./availability-page.component.sass'],
+})
+export class AvailabilityPageComponent implements OnInit {
+    // eslint-disable-next-line @angular-eslint/no-empty-lifecycle-method
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/availability-page/availability-page.component.ts]: Please do not use eslint-disable

File: frontend/package.json [hunk 2]
```diff
@@ -1,65 +1,65 @@
...
-        "karma-chrome-launcher": "^3.1.1",
-        "karma-coverage": "^2.1.1",
-        "karma-jasmine": "^4.0.2",
-        "karma-jasmine-html-reporter": "~1.7.0",
-        "prettier": "^2.7.1",
-        "prettier-eslint": "^15.0.1",
-        "typescript": "^4.6.4"
-    }
-}
+  "name": "frontend",
```
Comment (tatianahutii) [inline frontend/package.json]: Why did you remove all these spaces?

File: frontend/src/app/modules/availability/availability-page/availability-page.component.sass [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.container
+    width: 100vw
+    min-height: 100%
+    display: flex
+    flex-direction: column
+    align-items: center
+    background-color: #eef0f4
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/availability-page/availability-page.component.sass]: Move color to the shared file and use this value here

File: frontend/src/app/modules/availability/slot/slot.component.html [hunk 4]
```diff
@@ -0,0 +1,74 @@
+<mat-card class="card-container" *ngIf="this.isChecked">
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/slot/slot.component.html]: Why do you need 'this' here?

File: frontend/src/app/modules/availability/slot/slot.component.sass [hunk 5]
```diff
@@ -0,0 +1,74 @@
+hr
+    border: 0
+    border-top: 3px solid #eef0f4
+
+.card-container-disabled
+    opacity: 0.7
+
+.card-container
+    background-color: #fff
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/slot/slot.component.sass]: Move color to the shared file and use this value here

File: frontend/src/app/modules/availability/slot/slot.component.sass [hunk 6]
```diff
@@ -0,0 +1,74 @@
+hr
+    border: 0
+    border-top: 3px solid #eef0f4
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/slot/slot.component.sass]: Move color to the shared file and use this value here

File: frontend/src/app/modules/availability/slot/slot.component.sass [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+        flex-direction: column
+        justify-content: center
+        .slot-header
+            height: 10%
+            display: flex
+            align-items: center
+            justify-content: space-between
+            p
+                font-size: 0.8em
+                color: #d2d2d2
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/slot/slot.component.sass]: Move color to the shared file and use this value here

File: frontend/src/app/modules/availability/slot/slot.component.sass [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+.time-options
+    height: 10%
+    display: flex
+    align-items: flex-start
+    justify-content: flex-start
+    p
+        font-size: 1.2em
+        margin: 0.4em 0 0.4em 0
+        font-weight: bold
+        color: #687282
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/slot/slot.component.sass]: Move color to the shared file and use this value here

File: frontend/src/app/modules/availability/slot/slot.component.sass [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    display: flex
+    justify-content: space-between
+    align-items: center
+    font-size: 0.8em
+    margin: 1em 0 0 0
+    .link
+        display: flex
+        align-items: center
+        a
+            color: #808896
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/slot/slot.component.sass]: Move color to the shared file and use this value here

File: frontend/src/app/modules/availability/slot/slot.component.sass [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+        .icon
+            width: 18px
+            height: 18px
+            margin-right: 0.7em
+            filter: invert(94%) sepia(3%) saturate(1127%) hue-rotate(185deg) brightness(88%) contrast(86%)
+    .menu-icon
+        color: #a7a7a7
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/slot/slot.component.sass]: Move color to the shared file and use this value here

File: frontend/src/app/modules/availability/slot/slot.component.sass [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+        background-color: #fff
+        cursor: pointer
+        width: 18px
+        height: 18px
+
+.meeting-details
+    display: flex
+    flex-direction: column
+    p
+        color: #a7a7a7
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/slot/slot.component.sass]: Move color to the shared file and use this value here

File: frontend/src/app/modules/availability/slot/slot.component.ts [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import { Component, Input } from '@angular/core';
+import { MatSlideToggleChange } from '@angular/material/slide-toggle';
+
+@Component({
+    selector: 'app-slot',
+    templateUrl: './slot.component.html',
+    styleUrls: ['./slot.component.sass'],
+})
+export class SlotComponent {
+    @Input() public slot: {
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/slot/slot.component.ts]: Create a separate interface for this

File: frontend/src/app/modules/availability/team/team.component.sass [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    display: grid
+    grid-template-columns: repeat(auto-fill, 30%)
+    grid-gap: 1.2em
+
+.team-info
+    display: flex
+    align-items: flex-start
+    margin-top: 1em
+    .team-name
+        color: #808896
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/team/team.component.sass]: Move color to the shared file and use this value here

File: frontend/src/app/modules/availability/team/team.component.ts [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+            avatars: ['assets/bulochka.PNG'],
+        },
+        {
+            time: '15 min',
+            user: 'Me & 1 Booker',
+            link: 'Link.com/heorhii',
+            meetingPlace: 'Zoom',
+            avatars: ['assets/bulochka.PNG', 'assets/bulochka.PNG', 'assets/bulochka.PNG'],
+        },
+    ];
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/team/team.component.ts]: Move this to a separate file (helper)

File: frontend/src/app/modules/availability/team/team.component.ts [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+
+    // eslint-disable-next-line @angular-eslint/no-empty-lifecycle-method
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/team/team.component.ts]: Do not use eslint-disable-next

File: frontend/src/app/shared/styles/common-page-styles.sass [hunk 16]
```diff
@@ -1,223 +1,220 @@
...
+                i
+                    &::after
+                        transform: translate3d(16px, 2px, 0)
+
+    input
+        display: none
+        &:checked
+            +
+            i
+                background-color: #4C7DE7
```
Comment (tatianahutii) [inline frontend/src/app/shared/styles/common-page-styles.sass]: Move color to the shared file and use this value here

File: frontend/src/app/shared/styles/common-page-styles.sass [hunk 17]
```diff
@@ -1,223 +1,220 @@
...
-
+    background: #E1E1E1
+    text-align: center
+    padding: 7px
+    height: 26px
+    display: flex
+    align-items: center
+    justify-content: center
+    border-radius: 4px 0 0 4px
+    border: 1px solid #BEC5D0
```
Comment (tatianahutii) [inline frontend/src/app/shared/styles/common-page-styles.sass]: Move color to the shared file and use this value here

File: frontend/src/app/shared/styles/common-page-styles.sass [hunk 18]
```diff
@@ -1,223 +1,220 @@
...
+    p
+        font-size: 14px
+        color: #667080
```
Comment (tatianahutii) [inline frontend/src/app/shared/styles/common-page-styles.sass]: Move color to the shared file and use this value here

File: frontend/src/app/shared/styles/common-page-styles.sass [hunk 19]
```diff
@@ -1,223 +1,220 @@
...
-	border: 2px solid #BEC5D0
-	border-left: none
-	border-radius: 0 4px 4px 0
-	height: 36px
-	flex: none
-	order: 1
-	align-self: stretch
-	flex-grow: 0
-	width: 327px
+    background: #FFFFFF
```
Comment (tatianahutii) [inline frontend/src/app/shared/styles/common-page-styles.sass]: Move color to the shared file and use this value here

File: frontend/src/app/shared/styles/common-page-styles.sass [hunk 20]
```diff
@@ -1,223 +1,220 @@
...
+    border: 2px solid #BEC5D0
```
Comment (tatianahutii) [inline frontend/src/app/shared/styles/common-page-styles.sass]: Move color to the shared file and use this value here

File: frontend/src/app/shared/styles/common-page-styles.sass [hunk 21]
```diff
@@ -1,223 +1,220 @@
...
-	width: 100%
-	font-weight: 700
-	font-size: 12px
-	line-height: 20px
-	display: flex
-	align-items: center
-	text-align: center
-	justify-content: center
-	color: #667080
+    background: #FFFFFF
```
Comment (tatianahutii) [inline frontend/src/app/shared/styles/common-page-styles.sass]: Move color to the shared file and use this value here

File: frontend/src/app/shared/styles/common-page-styles.sass [hunk 22]
```diff
@@ -1,223 +1,220 @@
...
+    cursor: pointer
+    border: 2px solid #BEC5D0
```
Comment (tatianahutii) [inline frontend/src/app/shared/styles/common-page-styles.sass]: Move color to the shared file and use this value here

File: frontend/src/app/shared/styles/common-page-styles.sass [hunk 23]
```diff
@@ -1,223 +1,220 @@
...
-	box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.16)
-	border-radius: 4px
-	display: flex
-	flex-direction: row
-	align-items: center
-	justify-content: space-between
-	max-width: 360px
-	width: 100%
-	padding: 11px 23px
+    background: #FFFFFF
```
Comment (tatianahutii) [inline frontend/src/app/shared/styles/common-page-styles.sass]: Move color to the shared file and use this value here

File: frontend/src/styles/material.sass [hunk 24]
```diff
@@ -6,3 +6,21 @@
  *.mat-tab-group.mat-primary .mat-ink-bar
  *    background-color: red
  */
+.mat-menu-panel
+    border: 1px solid #687282
```
Comment (tatianahutii) [inline frontend/src/styles/material.sass]: Move color to the shared file and use this value here

File: frontend/src/styles/material.sass [hunk 25]
```diff
@@ -6,3 +6,21 @@
...
+    display: flex
+    justify-content: center
+
+.mat-menu-panel .mat-menu-item
+    padding-left: 5px
+    display: flex
+    flex-direction: row
+    align-items: center
+    font-size: 1em
+    color: #687282
```
Comment (tatianahutii) [inline frontend/src/styles/material.sass]: Move color to the shared file and use this value here

**Review by tatianahutii — CHANGES_REQUESTED** [review]

File: frontend/src/app/modules/availability/availability-page/availability-page.component.ts (hunk 1 above)
Comment (olena-kozak) [reply frontend/src/app/modules/availability/availability-page/availability-page.component.ts]: Can I remove 'implement OnInit' here? Because I don't use it for now.

File: frontend/src/app/modules/availability/slot/slot.component.sass (hunk 5 above)
Comment (olena-kozak) [reply frontend/src/app/modules/availability/slot/slot.component.sass]: Can I create separate file for colors, or I should add it to common-page-styles.sass?

File: frontend/src/app/modules/availability/availability-page/availability-page.component.ts (hunk 1 above)
Comment (tatianahutii) [reply frontend/src/app/modules/availability/availability-page/availability-page.component.ts]: Yes

File: frontend/src/app/modules/availability/slot/slot.component.sass (hunk 5 above)
Comment (tatianahutii) [reply frontend/src/app/modules/availability/slot/slot.component.sass]: It would be better to create a separate file
Something like color-variables

File: frontend/src/app/core/helpers/slots-data-helper.ts [hunk 26]
```diff
@@ -0,0 +1,27 @@
+import { Slot } from '@core/interfaces/slot-interface';
+
+export class SlotsDataHelper {
+    public static slots: Array<Slot> = [
```
Comment (tatianahutii) [inline frontend/src/app/core/helpers/slots-data-helper.ts]:
```suggestion
export const getDefaultSlots = () => [
```

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Comment by tatianahutii** [comment]
Could you please rework your file in this way?
image

And get rid of this file
https://github.com/BinaryStudioAcademy/bsa-2022-easymeets/blob/development/frontend/src/app/modules/user/color-variables.sass

File: frontend/src/app/modules/availability/team/team.component.ts [hunk 27]
```diff
@@ -0,0 +1,19 @@
...
+@Component({
+    selector: 'app-team',
+    templateUrl: './team.component.html',
+    styleUrls: ['./team.component.sass'],
+})
+export class TeamComponent {
+    public slots: Array<Slot>;
+
+    constructor() {
+        const { getDefaultSlots } = data;
```
Comment (tatianahutii) [inline frontend/src/app/modules/availability/team/team.component.ts]:
```suggestion
import  { getDefaultSlots }  from '@/core/helpers/slots-data-helper';

@Component({
    selector: 'app-team',
    templateUrl: './team.component.html',
    styleUrls: ['./team.component.sass'],
})
export class TeamComponent {
    public slots: Array<Slot>;

    constructor() {
```

**Review by tatianahutii — CHANGES_REQUESTED** [review]

**Review by tatianahutii — APPROVED** [review]
LGTM
